.github/workflows/ci.yml

```
@@ -271,14 +271,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        name: [system_deps, bundled_deps, system_deps_w_chisels]
+        name: [system_deps, bundled_deps, system_deps_w_chisels, asan]
         include:
           - name: system_deps
             cmake_opts: -DUSE_BUNDLED_DEPS=False
           - name: bundled_deps
             cmake_opts: -DUSE_BUNDLED_DEPS=True
           - name: system_deps_w_chisels
             cmake_opts: -DUSE_BUNDLED_DEPS=False -DWITH_CHISEL=True
+          - name: asan
+            cmake_opts: -DUSE_BUNDLED_DEPS=False -DCMAKE_C_FLAGS=-fsanitize=address -DCMAKE_CXX_FLAGS=-fsanitize=address
```

This is cool! I've been wanting to add it for some time now.

We might want to add `CMAKE_BUILD_TYPE=Debug` so we can get symbols for the ASAN reports. We could also add UBSAN, since it could catch some nasty undefined behaviour bugs and it works well with ASAN.
```suggestion
            cmake_opts: -DUSE_BUNDLED_DEPS=False -DCMAKE_C_FLAGS="-fsanitize=address -fsanitize=undefined" -DCMAKE_CXX_FLAGS="-fsanitize=address -fsanitize=undefined" -DCMAKE_BUILD_TYPE=Debug
```

We should also try to fix any memory leaks and/or undefined behavior that gets reported.

---

Historically I've not had a lot of luck with ubsan, I set up the tests locally so I can try it out

---

> We should also try to fix any memory leaks and/or undefined behavior that gets reported.

We can start from this one I think: https://github.com/falcosecurity/libs/pull/983 :)

---

Yes, I can confirm, UBSAN segfaults for me even on regular unit tests 😭 needs further investigation